### PR TITLE
Import HRQ into HNC (disabled by default)

### PR DESCRIPTION
**Pull request by adrianludwin**
This implementation of Hierarchical Resource Quotas (HRQs) is based on
the implementation from the GKE Hierarchy Controller.  It's not a
straight import since the GKE version was deployed as a separate
workload, but I've made minimal adaptations to get it to fit into HNC.

Tested: This has only been _very_ lightly tested in HNC so far. I've
imported all the integ tests and E2E tests, and they all pass, but I
certainly wouldn't recommend that anyone use it just yet. The default
manifest includes the HRQ CRD and permissions but nothing else (notably,
the HRQ webhooks), and one feature - the periodic resync of HRQ usages -
is not yet implemented (I'll do that in a followup PR).

### CODE REVIEW

**Review by rjbez17 — COMMENTED** [review]
I think I followed along and looks good to me, but it's pretty big. One thing that might be useful is some examples/docs, but I assume those will come shortly before the release?

File: cmd/manager/main.go [hunk 1]
```diff
@@ -149,6 +150,7 @@ func parseFlags() {
 	flag.Var(&managedNamespaceLabels, "managed-namespace-label", "A regex indicating the labels on namespaces that are managed by HNC. These labels may only be set via the HierarchyConfiguration object. All regexes are implictly wrapped by \"^...$\". This argument can be specified multiple times. See the user guide for more information.")
 	flag.Var(&managedNamespaceAnnots, "managed-namespace-annotation", "A regex indicating the annotations on namespaces that are managed by HNC. These annotations may only be set via the HierarchyConfiguration object. All regexes are implictly wrapped by \"^...$\". This argument can be specified multiple times. See the user guide for more information.")
 	flag.BoolVar(&webhooksOnly, "webhooks-only", false, "Disables the controllers so HNC can be run in HA webhook mode")
+	flag.BoolVar(&enableHRQ, "enable-hrq", false, "Enables hierarchical resource quotas")
```
[the remark this answers is not in the document]
Comment (adrianludwin) [reply cmd/manager/main.go:153]: Will do

**Comment by rjbez17** [comment]
Third time's the charm

**Pull request merged**